ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionsActivityStore.swift

```diff
@@ -227,6 +227,8 @@ class TransactionsActivityStore: ObservableObject, WalletObserverStore {
             $0.caseInsensitiveCompare(token.contractAddress) == .orderedSame
           })
         }
+      // remove duplicate
+      nftsWithoutMetadata = Array(Set(nftsWithoutMetadata))
```

Comment:
Shouldn't this just occur inside the helper function so the fix can apply elsewhere?
`func fetchNFTMetadata(tokens: [BraveWallet.BlockchainToken], ipfsApi: IpfsAPI) async -> [String: BraveWallet.NftMetadata]`

Reply:
i know you gonna comment on that once i pushed the change lol
yeah i will move it to `fetchNFTMetadata` haha

Reply:
updated at [99f8260](https://github.com/brave/brave-core/pull/26174/commits/99f8260f44088596dca2d95fe2549858b224438b)